src/bp/core/routers/admin/bots.ts

```diff
@@ -29,6 +29,7 @@ export class BotsRouter extends CustomRouter {
   private needPermissions: (operation: string, resource: string) => RequestHandler
   private hasPermissions: (req: RequestWithUser, operation: string, resource: string) => Promise<boolean>
   private assertBotpressPro: RequestHandler
+  private assertWorkspace = assertWorkspace()
```

Comment:
This property is not required, the reason we have needPermissions and hasPermissions here is so we can provide the workspace service to the middleware. You can simply use `assertWorkspace` instead of `this.assertWorkspace`

Reply:
I am afraid directly using it won't work, since it needs to be evaluated after setting the `.workspace` (e.g. in `checkTokenHeader`). So it is was made `async`, and set as a class member.

On directly using it (e.g. just in `auth.ts`) the whole admin page is not shown.

Reply:
If you change the signature of assert workspace from 
`assertWorkspace = () => async (req: RequestWithUser, _res: Response, next: NextFunction) =>`
to 
`assertWorkspace = async (req: RequestWithUser, _res: Response, next: NextFunction) =>`

then you can use it directly:
```js
router.get(
      '/me/profile',
      this.checkTokenHeader,
      assertWorkspace,
      this.asyncMiddleware(async (req: RequestWithUser, res) => {
```

Reply:
I should have thought about it!

As always, forgive my ignorance, and thanks for your patience.

Reply:
Absolutely no worry, we really appreciate your contributions!